lib/time/time.go

```diff
@@ -0,0 +1,454 @@
+}
+
+func durationminutes(fnname string, recV starlark.Value, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	recv := gotime.Duration(recV.(Duration))
+	return starlark.Float(recv.Minutes()), nil
+}
+
+func durationseconds(fnname string, recV starlark.Value, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	recv := gotime.Duration(recV.(Duration))
+	return starlark.Float(recv.Seconds()), nil
+}
+
+func durationnanoseconds(fnname string, recV starlark.Value, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	recv := gotime.Duration(recV.(Duration))
+	return starlark.MakeInt64(recv.Nanoseconds()), nil
+}
+
+// Time is a starlark representation of a point in time
+type Time gotime.Time
+
+// String implements the Stringer interface
+func (t Time) String() string { return gotime.Time(t).String() }
+
+// Type returns a short string describing the value's type.
+func (t Time) Type() string { return "time" }
+
+// Freeze renders time immutable. required by starlark.Value interface
+// because Time is already immutable this is a no-op
+func (t Time) Freeze() {}
+
+// Hash returns a function of x such that Equals(x, y) => Hash(x) == Hash(y)
+// required by starlark.Value interface
+func (t Time) Hash() (uint32, error) { return hashString(t.String()), nil }
+
+// Truth returns the truth value of an object required by starlark.Value interface
+func (t Time) Truth() starlark.Bool { return starlark.Bool(gotime.Time(t).IsZero()) }
+
+// Attr gets a value for a string attribute, implementing dot expression support in starklark. required by starlark.HasAttrs interface
+func (t Time) Attr(name string) (starlark.Value, error) { return builtinAttr(t, name, timeMethods) }
+
+// AttrNames lists available dot expression strings for time. required by starlark.HasAttrs interface
+func (t Time) AttrNames() []string { return builtinAttrNames(timeMethods) }
+
+// CompareSameType implements comparison of two Time values. required by starlark.Comparable interface
+func (t Time) CompareSameType(op syntax.Token, yV starlark.Value, depth int) (bool, error) {
+	x := gotime.Time(t)
+	y := gotime.Time(yV.(Time))
+	cp := 0
+	if x.Before(y) {
+		cp = -1
+	} else if x.After(y) {
+		cp = 1
+	}
+	return threeway(op, cp), nil
+}
+
+// Binary implements binary operators, which satisfies the starlark.HasBinary interface
+func (t Time) Binary(op syntax.Token, yV starlark.Value, side starlark.Side) (starlark.Value, error) {
+	x := gotime.Time(t)
+
+	switch yV.(type) {
+	case Duration:
+		y := gotime.Duration(yV.(Duration))
+		switch op {
+		// time + duration = time
+		case syntax.PLUS:
+			return Time(x.Add(y)), nil
+		// time - duration = time
+		case syntax.MINUS:
+			return Time(x.Add(-y)), nil
+		}
+	case Time:
+		y := gotime.Time(yV.(Time))
+		switch op {
+		// time - time = duration
+		case syntax.MINUS:
+			if side == starlark.Left {
+				return Duration(x.Sub(y)), nil
+			}
+			return Duration(y.Sub(x)), nil
```

Comment:
Use if { return E1 } else { return E2 }  when the two E expressions have parallel structure.

(It's fine to omit the second else if the first case is an early return.)